### PR TITLE
REP-62 review notifications

### PR DESCRIPTION
**Pull request by edwh**
Notify regional admins when a business is set ready for review.

This introduces email notifications in a similar way to restarters.

There's some c&p code from DoctrineBusinessRepository into DoctrineUserRepository - consider whether you want me to rationalise this.

### CODE REVIEW

File: src/Application/Commands/Business/ImportFromHttpRequest/ImportFromHttpRequestHandler.php [hunk 1]
```diff
@@ -93,7 +109,28 @@ public function handle(ImportFromHttpRequestCommand $command)
...
+        if ($currentPublishingStatus !== PublishingStatus::READY_FOR_REVIEW && $newPublishingStatus === PublishingStatus::READY_FOR_REVIEW
+            && auth()->user()->isEditor()
+        ) {
+            // We have published the business as an Editor.  We want to alert regional admin.  At the moment
+            // editors are not restricted to a region and therefore we alert all of them.
+            $admins = $this->userRepository->findBy([
+                [
+                    'field' => 'repairDirectoryRole',
+                    'operator' => Operators::EQUAL,
+                    'value' => 2
```
Comment (ngm) [inline src/Application/Commands/Business/ImportFromHttpRequest/ImportFromHttpRequestHandler.php:126]: Hmm, I guess we don't have a constant for the role ids anywhere.  Should we add them to the Role class?

**Review by ngm — CHANGES_REQUESTED** [review]
All looks good - just the one change with the user name used in the email.

Not fussed about the duplication between the repositories - not worth our time wrestling with abstracting them, certainly not now, and not if we hope to move back to Eloquent in some future utopia..

File: app/Notifications/AdminNewBusinessReadyForReview.php [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+     */
+    public function toMail($notifiable)
+    {
+        return (new MailMessage)
+            ->subject(__('notification.business_ready_for_review_subject', [
+                'business' => $this->business->getName()
+            ]))
+            ->greeting(__('notification.greeting'))
+            ->line(__('notification.business_ready_for_review_body', [
+                'by' => $notifiable->getName(),
```
[the remark this answers is not in the document]
Comment (edwh) [reply app/Notifications/AdminNewBusinessReadyForReview.php]: Ah yes.  Fix applied.

File: src/Application/Commands/Business/ImportFromHttpRequest/ImportFromHttpRequestHandler.php (hunk 1 above)
Comment (edwh) [reply src/Application/Commands/Business/ImportFromHttpRequest/ImportFromHttpRequestHandler.php]: I've added some, though some of the code uses a different method - see for example User::isRegionalAdmin.  I've not changed that as part of this, though - let me know if you want me to.

**Review by ngm — APPROVED** [review]

File: src/Application/Commands/Business/ImportFromHttpRequest/ImportFromHttpRequestHandler.php (hunk 1 above)
Comment (ngm) [reply src/Application/Commands/Business/ImportFromHttpRequest/ImportFromHttpRequestHandler.php]: No that's OK - thanks for adding it here, more readable now.

**Pull request merged**